docs/plugins/create.md

```diff
@@ -1,104 +1,525 @@
+| `ASDF_CONCURRENCY`       | the number of cores to use when compiling the source code. Useful for setting `make -j` |
+| `ASDF_DOWNLOAD_PATH`     | the path to where the source code or binary was downloaded to by `bin/download`         |
+| `ASDF_PLUGIN_PATH`       | the path the plugin was installed                                                       |
+| `ASDF_PLUGIN_SOURCE_URL` | the source URL of the plugin                                                            |
+| `ASDF_PLUGIN_PREV_REF`   | prevous `git-ref` of the plugin repo                                                    |
+| `ASDF_PLUGIN_POST_REF`   | updated `git-ref` of the plugin repo                                                    |
+| `ASDF_CMD_FILE`          | resolves to the full path of the file being sourced                                     |
+
+::: tip NOTE
+
+**Not all environment variables are available in all scripts.** Check the
+documentation for each script below to see which env vars are available to it.
+
+:::
 
 ## Required Scripts
 
-- `bin/list-all` - lists all installable versions
-- `bin/download` - download source code or binary for the specified version
-- `bin/install` - installs the specified version
+### `bin/list-all` <Badge type="tip" text="required" vertical="middle" />
 
-## Environment Variables
+**Description**
 
-All scripts except `bin/list-all` will have access to the following env vars to act upon:
+List all installable versions.
 
-- `ASDF_INSTALL_TYPE` - `version` or `ref`
-- `ASDF_INSTALL_VERSION` - if `ASDF_INSTALL_TYPE` is `version` then this will be the version number. Else it will be the git ref that is passed. Might point to a tag/commit/branch on the repo.
-- `ASDF_INSTALL_PATH` - the dir where it _has been_ installed (or _should_ be installed in case of the `bin/install` script)
+**Output Format**
 
-These additional environment variables will be available to the `bin/install` script:
+Must print a string with a **space-separated** list of versions. For example:
 
-- `ASDF_CONCURRENCY` - the number of cores to use when compiling the source code. Useful for setting `make -j`.
-- `ASDF_DOWNLOAD_PATH` - the path to where the source code or binary was downloaded by the `bin/download` script.
+```text:no-line-numbers
+1.0.1 1.0.2 1.3.0 1.4
+```
 
-These additional environment variables will be available to the `bin/download` script:
+Newest version should be last.
 
-- `ASDF_DOWNLOAD_PATH` - the path to where the source code or binary should be downloaded.
+asdf core will print each version on its own line, potentially pushing some
+versions offscreen.
 
-#### bin/list-all
+**Sorting**
 
-Must print a string with a space-separated list of versions. Example output would be the following:
+If versions are being pulled from releases page on a website it's recommended to
+leave the versions in the provided order as they are often already in the
+correct order. If they are in reverse order piping them through `tac` should
```

Comment:
Is `tac` available everywhere?

Reply:
As far as I know `tac` is part of the GNU coreutils, so no, not available everywhere, but somewhat common. This is a statement I carried across from the older docs.

There are definite improvements to be made here. I pushed this through to just get the updates out there. Iterative improvements are welcome.